pkg/operator/customroute/custom_route_controller.go

```diff
@@ -0,0 +1,292 @@
+package customroute
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	configsetterv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	routeclient "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+	routeinformer "github.com/openshift/client-go/route/informers/externalversions/route/v1"
+	routev1lister "github.com/openshift/client-go/route/listers/route/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/openshift/library-go/pkg/certs"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"github.com/openshift/library-go/pkg/route/routecomponenthelpers"
+)
+
+// customRouteController applies the given route asset.
+// It reads its component route, if configured, from the cluster ingress configuration
+// and synchronizes the component route's custom CA serving cert secret to the given destination.
```

Comment:
nit: just `custom serving cert secret to the given destination`, my mind stumbles on the `custom CA` part

Reply:
agreed, let's fix